examples/known-problems/verifythis/2019/challenge2b.pvl

```diff
@@ -1,120 +1,120 @@
-   && left!= null && right!=null&&parent!=null;
-  
-  
-  yields int[] parent;
-  yields int[] left;
-  yields int[] right;
-  yields int root;
-  ensures right!=null;
-  ensures parent!=null;
-  ensures left!= null;
-  ensures s.length == parent.length && s.length == left.length && s.length == right.length;
-  ensures (\forall* int i;0<=i && i < s.length; Perm(left[i],1) **  Perm(right[i],1)**  Perm(parent[i],1));  
-  ensures correctness_invar(s,parent,left,right);
-  ensures (\forall int i;0<=i && i < s.length; parent[i]!=-1 ==> 0 <= parent[i] && parent[i] < s.length);
-  ensures correctness_upto(s,parent,left,right,root,s.length);
-  cartTree(seq<int> s){
-    
-    parent = new int[s.length];
-    left = new int[s.length];
-    right = new int[s.length];
-    int root = -1;
-    int i=0;
-    
-    loop_invariant 0 <= i && i <= s.length;
-    loop_invariant parent.length == s.length;
-    loop_invariant left.length == s.length;
-    loop_invariant right.length == s.length;
-    loop_invariant left!=null && right!=null && parent!=null;
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(parent[j],1));
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(left[j],1));
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(right[j],1));
-    while(i<s.length){
-      left[i] = -1;
-      right[i] = -1;
-    }
-    
-    i = 0;
-    loop_invariant 0 <= i && i <= s.length;
-    loop_invariant parent.length == s.length;
-    loop_invariant left.length == s.length;
-    loop_invariant right.length == s.length;
-    loop_invariant left!=null && right!=null && parent!=null;
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(parent[j],1));
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(left[j],1));
-    loop_invariant (\forall* int j;0 <= j && j < s.length; Perm(right[j],1));
-    loop_invariant correctness_invar(s,parent,left,right);
-    loop_invariant correctness_upto(s,parent,left,right,root,i);
-    while(i<s.length){
-      if(rightNeighbor(s,i)==s.length){
-        if(leftNeighbor(s,i)==-1){
-          parent[i] = -1;
-          root = i;
-        }else{
-          parent[i] = leftNeighbor(s,i); // not -1
-          right[leftNeighbor(s,i)] = i;
-        }
-      } else {
-        if(leftNeighbor(s,i)==-1){
-          parent[i] = rightNeighbor(s,i); // not s.length
-          left[rightNeighbor(s,i)] = i;
-        }else{
-          if (s[leftNeighbor(s,i)] > s [rightNeighbor(s,i)]){
-            parent[i] = leftNeighbor(s,i); // not -1
-            right[leftNeighbor(s,i)] = i;
-          }else{
-            parent[i] = rightNeighbor(s,i); // not s.length
-            left[rightNeighbor(s,i)] = i;
-          }
-        }
-      }
-      i++;
-    }
-  }
-}
+// -*- tab-width:2 ; indent-tabs-mode:nil -*-
+//:: cases VerifyThis19TwoB
+//:: tools silicon
+//:: suite medium problem-fail
```

Comment:
Why is this entire file changed, even though you seem to only change the suite? It might require some git hacking to drop the unnecessary changes... Maybe the newlines were changed accidentally?

Reply:
I am not sure why it is showing it in this way. However git blame in the IDE is working correctly.